.github/workflows/ci.yml

```diff
@@ -19,11 +17,37 @@ env:
   DART_VERSION: '2.14.4'
   DART_WORKSPACE: ${{ github.workspace }}/discovery_engine
   RUST_WORKSPACE: ${{ github.workspace }}/discovery_engine_core
+  RUST_NIGHTLY: nightly-2021-09-09
+  RUSTFLAGS: "-D warnings"
+  RUSTDOCFLAGS: "-D warnings"
 
 
 jobs:
+  selection:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      dart: ${{ steps.filter.outputs.dart }}
+      rust: ${{ steps.filter.outputs.rust }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v2.10.2
+        id: filter
+        with:
+          filters: |
+            dart:
+              - 'discovery_engine/**'
+              - 'discovery_engine_core/**'
```

Comment:
Spoke to @rustonaut  and he explained me that: 
`discovery_engine_core/` generates a C header which is then used by `discovery_engine` to generate dart code using a external tool (which happend to have been written in rust, but it's  in another repo). Through for testing dart we need to fully build rust.
`discovery_engine_flutter` will only have dart code? Yes (ignoring the kotlin/swift/etc. glue code for compiling it as app, in which case you also need fully build rust code, testing it also needs to have rust build as it might call functions implemented in rust during tests).

Reply:
Ok, we can see if we can do something better later with a cache for generated file bu is also a bit risky.